Use a Python virtual environment

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request updates the Docker image configuration to use a Python virtual environment.

> [!NOTE]
> I am creating this pull request against a branch that will represent a larger body of work. This branch will later have a pull request against `develop` once the body of work is completed.

<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Using virtual environments is a Python best practice. It will also ensure our Python dependencies are isolated from the system Python environment.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Prefer calling pip as a module

Instead of relying on `pip3` being on the PATH we instead call the
module through the Python executable. This ensures that the `pip` being
used is in the same environment as the `python3` being used.

mcdonnnj committed:
Use a Python virtual environment in the Docker image

Using a virtual environment is a Python best practice. We also
consolidate all of the Python dependency installation steps into a
single RUN instruction. This ensures that Python setup is cached in one
layer and mirrors the logical organization of this being a single step.

jsf9k commented on Dockerfile:
```diff
@@ -42,25 +43,32 @@ RUN addgroup --system --gid ${CISA_GID} ${CISA_GROUP} \
+# because the ln binary in Alpine Linux does not support long flags.
 ###
-RUN pip3 install --no-cache-dir https://github.com/cisagov/skeleton-python-library/archive/v${VERSION}.tar.gz
+RUN ln -sf "$(command -v python3)" "${VIRTUAL_ENV}"/bin/python3
```
Nice trick!

Please add a short comment explaining what the `-s` and `-f` flags do since we cannot use long options here.

mcdonnnj replied:
Please let me know if [77b5e34](https://github.com/cisagov/skeleton-docker/pull/190/commits/77b5e3483d2063358c20fe5b04648a9c777a8bdd) works for you.

jsf9k reviewed (changes requested):
One minor ask, otherwise this looks great!

mcdonnnj committed:
Explain `ln` options being used

Since we cannot use long options on Alpine Linux we should explain what
the short options we are using do. I also changed the order of options
so that they are in alphabetical order.

Co-authored-by: Shane Frasier <jeremy.frasier@gwe.cisa.dhs.gov>

dav3r reviewed (approved):
Very nice! 👍
